util/util.go

```diff
@@ -372,6 +391,42 @@ func indexOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerRefe
 	return -1
 }
 
+// PointsTo returns true if any of the owner references point to the given target
+// Deprecated: Use IsOwnedByObject to cover differences in API version or backup/restore that changed UIDs.
+func PointsTo(refs []metav1.OwnerReference, target *metav1.ObjectMeta) bool {
+	for _, ref := range refs {
+		if ref.UID == target.UID {
+			return true
+		}
+	}
+	return false
+}
+
+// IsOwnedByObject returns true if any of the owner references point to the given target.
+func IsOwnedByObject(obj metav1.Object, target object) bool {
+	for _, ref := range obj.GetOwnerReferences() {
+		ref := ref
+		if refersTo(&ref, target) {
+			return true
+		}
+	}
+	return false
+}
+
+type object interface {
+	metav1.Object
+	GroupVersionKind() schema.GroupVersionKind
+}
```

Comment:
If we rebase on master, we should be able to consume `controllerutil.Object` from https://github.com/kubernetes-sigs/controller-runtime/pull/898